### DISCUSSION TITLE
一加13 Color os 15 打不开gkd的无障碍

### DISCUSSION THREAD

**Discussion by Paradox98** [gkd-kit/gkd#773] [Q&A] [answered]
### 日志文件

[log-1731390389757.zip](https://github.com/user-attachments/files/17711694/log-1731390389757.zip)

### BUG描述(文字/截图/视频)

coloros 15 root了 magisk28 lsposd 1.10.1 gkd 1.9.1
给了root权限 给了shizuku权限 点无障碍没反应

### 期望行为(文字/截图/视频)

能够正常使用无障碍

### 实际行为(文字/截图/视频)

无障碍点了没有报错但是就是打不开

https://github.com/user-attachments/assets/46a82fff-ab84-4e01-b0c7-8096956f6cdb

**Comment by lisonge** [comment]
> 点无障碍没反应

录个视频看看

**Comment by lisonge** [comment]
我不知道你的设备是怎么回事

从你的日志中我没有看到任何无障碍启动的日志

你可以试试 重装 GKD 或者重启手机

---

> 给了root权限 给了shizuku权限

都有这个权限了为什么不在 GKD 里面直接打开？

**Comment by yoholo** [comment]
同一加13，root
软件给root权限。
进设置无障碍打不开，只有之前降级后无障碍中提示未运行（具体内容不是很确定）才可以打开这个按钮。
后来又升级了1.9.1正式版，现在降级1.8.0、1.7.3也无法打开。卸载重装也无法打开。卸载重启，安装重启等都试过。没用。

**Comment by lisonge** [comment]
> 软件给root权限

GKD 不使用 root 权限持久运行，给了也没用

---

授予 写入安全设置权限 后 GKD 自身是可以自动启动无障碍的，不需要用户手动在系统无障碍设置里打开

没懂你们是为什么要在系统设置里去打开无障碍呢？

---

有没有可能是 https://gkd.li/guide/faq#restriction 这个原因呢？

**Comment by yoholo** [comment]
授予 写入安全设置权限？
没看到过有 受限制的设置 弹窗过
应用详情也没有允许受限制的设置项
设置里也找不到相关内容，包括关键词安全和限制，开发者设置里面禁止权限监控打开也没用
![437a7a69f69e1b61abb2434884c83a4](https://github.com/user-attachments/assets/0540a10e-e6a0-41ab-9fb2-7326df3d2659)

**Reply by yoholo** [reply]
root 一加8T colorOS13 应用详情也没有允许受限制的设置项，自动升级，完美运行

root 一加9 colorOS14 有受限制的设置弹窗，应用详情有允许受限制的设置项，打开后完美运行。

**Reply by yoholo** [reply]
![Screenshot_2024-11-13-07-07-27-35_befd3b55c4270226022f86e9f9cafb33](https://github.com/user-attachments/assets/768890c1-f011-4f1a-b5ec-7e094bf3c687)
root权限就是为了推荐的高级授权

**Reply by yoholo** [reply]
显示的是有权限，实际运行不了
问题和楼主一样，应用界面点了没反应，进无障碍也打不开
![Screenshot_2024-11-13-07-38-35-84_befd3b55c4270226022f86e9f9cafb33](https://github.com/user-attachments/assets/b5706d5f-cf4a-4bda-8ebb-bb4535d2a3b9)

**Reply by GuiltyGeass** [reply]
我一加13遇到同样问题，最开始我是点的了授权的，有一次gkd软件升级之后就开不了，后面重装，重启都不行。不止gkd，有别的应用我也遇到点不开无障碍的事情。不知道是不是coloros15本身的系统bug

**Reply by robot-error** [reply]
我也是这样

**Comment by namexiaolu** [comment]
卸载，然后用手机自带的文件管理器安装gkd，就可以了。
来自酷安网友的解决方案，亲测可用。

**Reply by yoholo** [reply]
亲测可行。。。停用了LSPOSED，有LuckyTool，替换了系统的应用安装程序。再用文件管理器安装后一切正常。之前都是浏览器和自动更新的。

**Reply by ZenQy** [reply]
> 卸载，然后用手机自带的文件管理器安装gkd，就可以了。 来自酷安网友的解决方案，亲测可用。

感谢，我也遇到了同样的问题，该用mt管理器安装，成功解决

**Comment by whsiny** [comment]
刚刚升级1.9.2也出现打不开无障碍，回退到1.9.2-beta.1正常启用。

**Reply by lisonge** [reply]
所以是为什么打不开无障碍？可以把日志和打不开原因的截图贴上来

只发一句打不开，谁都不知道原因

**Reply by whsiny** [reply]
刚刚重装了1.9.2想看看日志的，发现无障碍自动打开了。下午也和楼主一样点无障碍没反应。

**Comment by lisonge** [comment]
https://github.com/gkd-kit/gkd/issues/792#issuecomment-2498063120

**Comment by lisonge** [comment, marked as answer]
https://gkd.li/guide/faq#unable_open_a11y

**Reply by lisonge** [reply]
这个问题应该在 https://github.com/gkd-kit/gkd/releases/tag/v1.9.3-beta.1 下得到解决

**Comment by YamatoRyou** [comment]
补充一些**特殊用例**的测试结果.
机型与系统: 一加 9RT (Oxygen OS 15, 基于 Android 15, 非官方移植, Root 方案为 APatch)
GKD 版本: 1.10.1

----
1. 系统存在多用户, 且 GKD 与需要去广告的应用都安装在副用户环境;
2. 如果用 Root Explorer; 系统自带的文件管理应用或系统自带的浏览器安装 GKD, 则即使执行了以下命令也仍然无法正常打开无障碍权限. 而 GKD 却会提示已持有无障碍权限;
```
# 在副用户环境下加上 --user 10 以作用于指定用户 (10 为该用户的 UID)
appops set --user 10 li.songe.gkd ACCESS_RESTRICTED_SETTINGS allow
pm grant --user 10 li.songe.gkd android.permission.WRITE_SECURE_SETTINGS
```
3. 执行以下命令以撤销 GKD 的特殊权限, 用于消除测试期间的变量;
```
# 在副用户环境下加上 --user 10 以作用于指定用户 (10 为该用户的 UID)
appops set --user 10 li.songe.gkd ACCESS_RESTRICTED_SETTINGS deny
pm revoke --user 10 li.songe.gkd android.permission.WRITE_SECURE_SETTINGS
```
4. 最终解决方法是通过 scrcpy 为副用户安装 GKD, 而无需通过命令授予特殊权限.

----
主用户:
不限制安装来源, 于 GKD 安装后只需执行以下命令即可:
```
appops set li.songe.gkd ACCESS_RESTRICTED_SETTINGS allow
```

**Comment by moqiu365** [comment]
使用自带的文件管理安装GKD就可以打开了

**Reply by popokcn** [reply]
挖坟吗，哈哈哈，我说怎么是2024年的帖子。
